File: app/src/main/java/com/konektedi/vs/Motions/Opinions/OpinionsDataFactory.java
package com.konektedi.vs.Motions.Opinions;

import android.arch.lifecycle.MutableLiveData;
import android.arch.paging.DataSource;

import java.util.concurrent.Executor;

public class OpinionsDataFactory extends DataSource.Factory<Integer, Opinions> {

    MutableLiveData<OpinionsDataSource> mutableLiveData;
    OpinionsDataSource dataSource;
    Executor executor;
    private int motion_id;


    public OpinionsDataFactory(Executor executor, int motion_id) {
        this.mutableLiveData = new MutableLiveData<>();
        this.executor = executor;
        this.motion_id = motion_id;
    }


    @Override
    public DataSource create() {
        dataSource = new OpinionsDataSource(executor, motion_id);
        mutableLiveData.postValue(dataSource);
        return dataSource;
    }

    public MutableLiveData<OpinionsDataSource> getMutableLiveData() {
        return mutableLiveData;
    }

}

File: app/src/main/java/com/konektedi/vs/opinions/OpinionsDataFactory.kt
package com.konektedi.vs.opinions

import androidx.lifecycle.MutableLiveData
import androidx.paging.DataSource
import com.konektedi.vs.utilities.models.Opinion
import java.util.concurrent.Executor

class OpinionsDataFactory(private val executor: Executor, private val categoryId: Int) : DataSource.Factory<Int, Opinion>() {

    val mutableLiveData = MutableLiveData<OpinionsDataSource>()

    override fun create(): DataSource<Int, Opinion> {
        val dataSource = OpinionsDataSource(executor,categoryId)
        mutableLiveData.postValue(dataSource)
        return dataSource
    }
}